fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -257,6 +257,171 @@ sebasedBiasFieldMNI="$SubjectFolder/$AtlasSpaceFolder/Results/$NameOffMRI/${Name
+  log_Warn "  motion and may also disrupt data quality measures as shown in Power et al 2017 PLOS One "
+  log_Warn "  'Temporal interpolation alters motion in fMRI scans: Magnitudes and consequences for"
+  log_Warn "  artifact detection.' Slice timing correction and motion correction would ideally be performed"
+  log_Warn "  simultaneously; however, this is not currently supported by any major software tool. HCP-Style"
+  log_Warn "  fast TR fMRI data acquisitions (TR<=1s) avoid the need for slice timing correction, provide"
+  log_Warn "  major advantages for fMRI denoising, and are recommended."
+fi
+
+
+# -- Use of nonstandard BOLD mask
+
+if [ "${BOLDMask}" != 'T1_fMRI_FOV' ]; then
+  if [ "${BOLDMask}" != "T1_DILATED_fMRI_FOV" ] && [ "${BOLDMask}" != "T1_DILATED2x_fMRI_FOV" ] && [ "${BOLDMask}" != "fMRI_FOV" ] ; then
+    log_Err_Abort "--boldmask=${BOLDMask} is invalid! Valid options are: T1_fMRI_FOV (default), T1_DILATED_fMRI_FOV, T1_DILATED2x_fMRI_FOV, fMRI_FOV."
+  fi
+  ComplianceMsg+=" --boldmask=${BOLDMask}"
+  Compliance="LegacyStyleData"
+fi
+
+# -- Use of external BOLD reference
+
+if [ "$fMRIReference" = "NONE" ]; then
+  fMRIReferenceReg="NONE"    
+  fMRIReferencePath="NONE"
+  ReferenceResultsFolder="NONE"
+else
+  fMRIReferenceReg=`opts_DefaultOpt $fMRIReferenceReg "linear"`
+
+  # set reference and check if external reference (if one is specified) exists 
+
+  fMRIReferencePath="$Path"/"$Subject"/"$fMRIReference"
+  log_Msg "Using reference image from ${fMRIReferencePath}"
+  fMRIReferenceImage="$fMRIReferencePath"/"$ScoutName"_gdc
+  fMRIReferenceImageMask="$fMRIReferencePath"/"$ScoutName"_gdc_mask
+  ReferenceResultsFolder="$Path"/"$Subject"/"$AtlasSpaceFolder"/"$ResultsFolder"/"$fMRIReference"
+
+  if [ "$fMRIReferencePath" = "$fMRIFolder" ] ; then
+    log_Err_Abort "Specified BOLD reference (--fmriref=${fMRIReference}) is the same as the current BOLD (--fmriname=${NameOffMRI})!"
+  fi
+
+  if [ `${FSLDIR}/bin/imtest ${fMRIReferenceImage}` -eq 0 ] ; then
+    log_Err_Abort "Intended BOLD Reference does not exist (${fMRIReferenceImage})!"
+  fi 
+
+  if [ `${FSLDIR}/bin/imtest ${fMRIReferenceImageMask}` -eq 0 ] ; then
+    log_Err_Abort "Intended BOLD Reference mask does not exist (${fMRIReferenceImageMask})!"
+  fi 
+
+  if [ ! -d "$ReferenceResultsFolder" ] ; then
+    log_Err_Abort "Reference results folder does not exist and can not be used (${ReferenceResultsFolder})!"
+  fi 
+
+  # print warning
+
+  log_Warn "You are using an external reference (--fmriref=${fMRIReference}) for motion registration and"
+  log_Warn "  distortion correction and registration to T1w image. Please consider using this option only"
+  log_Warn "  in cases when only one BOLD Reference image is available or when processing low spatial"
+  log_Warn "  resolution legacy BOLD images (as in this cases within BOLD registration might be more robust" 
```

Comment:
Minor grammatical tweak:
`log_Warn " resolution legacy BOLD images (for which within BOLD registration might be more robust"`

Reply:
Done.